syntax = "proto3";

message GorasSentence {
    int32 id = 1;
    string speaker = 2;
    double when = 3;
    string to_who = 4;
    oneof content {
        GorasQuery query = 5;
        GorasAck ack = 6;
        GorasInform inform = 7;
        GorasCommand command = 8;
    }
}

message GorasQuery {
    string question = 1;
}

message GorasAck {
    int32 ref_id = 1;
    string comment = 2;
}

message GorasInform {
    string subject = 1;
    string verb = 2;
    string object = 3;
    string adjective = 4;
}

message GorasCommand {
    string subject = 1;
    string action = 2;
}